### PR TITLE
creates queues from config

### PR DESCRIPTION
**Pull request by luxe**
Simplify the startup of the shard backplane by providing config to queue creation functions.

### CODE REVIEW

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 1]
```diff
@@ -505,18 +505,34 @@ private void startFailsafeOperationThread() {
...
   @Override
   public void start(String clientPublicName) throws IOException {
-
     // Construct a single redis client to be used throughout the entire backplane.
     // We wish to avoid various synchronous and error handling issues that could occur when using
     // multiple clients.
     client = new RedisClient(jedisClusterFactory.get());
 
+    this.prequeue = createPrequeue(config);
+    this.operationQueue = createOperationQueue(config);
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:514]: No need for `this.` if it doesn't shadow a parameter.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 2]
```diff
@@ -505,18 +505,34 @@ private void startFailsafeOperationThread() {
...
+    if (config.getSubscribeToBackplane()) {
+      startSubscriptionThread();
+    }
+    if (config.getRunFailsafeOperation()) {
+      startFailsafeOperationThread();
+    }
+
+    // Record client start time
+    client.call(
+        jedis -> jedis.set("startTime/" + clientPublicName, Long.toString(new Date().getTime())));
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:525]: I see that this was just moved, but it really shouldn't be set without an expiration, and the timestamp acquisition is pretty awkward - why not just make it human readable?

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 3]
```diff
@@ -505,18 +505,34 @@ private void startFailsafeOperationThread() {
...
+  }
+
+  BalancedRedisQueue createPrequeue(RedisShardBackplaneConfig config) throws IOException {
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:528]: This looks like it qualifies for static

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java [hunk 4]
```diff
@@ -505,18 +505,34 @@ private void startFailsafeOperationThread() {
...
-    this.prequeue =
-        new BalancedRedisQueue(
-            config.getPreQueuedOperationsListName(),
-            getQueueHashes(config.getPreQueuedOperationsListName()));
+    return new BalancedRedisQueue(
+        config.getPreQueuedOperationsListName(),
+        getQueueHashes(config.getPreQueuedOperationsListName()));
+  }
 
+  OperationQueue createOperationQueue(RedisShardBackplaneConfig config) throws IOException {
```
Comment (werkt) [inline src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:535]: Likewise qualified for static.

**Review by werkt — APPROVED** [review]

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 1 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:514]: fixed

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 3 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:528]: changed the "hash generation" func to take the client as an argument, so these "queue creation" functions can also take the client and now be marked static.  
that's better.  when things are marked static they become obvious candidates to move out of the backplane in the future.

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 4 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:535]: fixed

File: src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java (hunk 2 above)
Comment (luxe) [reply src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java:525]: maybe the format is best suited for bf-admin? let me see about changing. @80degreeswest is it safe to expire this key?

**Pull request merged**